var/spack/repos/builtin/packages/python/package.py

```diff
@@ -953,9 +955,11 @@ def setup_dependent_run_environment(self, env, dependent_spec):
         # dependent_spec and prepend it to PYTHONPATH
         if dependent_spec.package.extends(self.spec):
             for lib in ['lib', 'lib64']:
```

Comment:
factor into a function?
```
def _get_python_paths(prefix, version):
    paths = (join_path(prefix,
                       lib,
                       'python' + str(version.up_to(2)),
                       'site-packages')
             for lib in ('lib', 'lib64'))
            ret.append(pth)
    return (pth for pth in if os.path.exists(pth))
```
then 
`python_paths.extend(_get_python_paths(d.prefix, self.version))`
and
```
for pth in _get_python_paths(dependent_spec.prefix, self.version)
    end.prepend_path(pth)  # maybe env has a better method for this?
```

Reply:
Changed to similar implementation.